os/fs/vfs/fs_open.c

```diff
@@ -115,18 +119,18 @@ int open(const char *path, int oflags, ...)
 	int ret;
 	int fd;
 
+	if (path == NULL) {
+		return -EINVAL;
+	}
+
 #ifdef CONFIG_FILE_MODE
 #ifdef CONFIG_CPP_HAVE_WARNING
 #warning "File creation not implemented"
 #endif
 
-	/* open() is a cancellation point */
-	(void)enter_cancellation_point();
-
 	/* If the file is opened for creation, then get the mode bits */
 
 	if ((oflags & (O_WRONLY | O_CREAT)) != 0) {
-		va_list ap;
 		va_start(ap, oflags);
```

Comment:
‘va_start’ was used in function vopen with fixed args.
I think we should use va_arg directly.
@junmin-kim